app/controllers/resources_controller.rb

```diff
@@ -59,20 +59,48 @@ def check_permission
     privilege = params[:privilege]
     raise ArgumentError, "privilege" unless privilege
 
-    result = assumed_role.allowed_to?(privilege, resource)
+    begin
+      result = assumed_role.allowed_to?(privilege, resource)
+
+      audit_success(privilege, result)
+    rescue => e
+      audit_failure(e, privilege)
+
+      raise
+    end
 
+    head(result ? :no_content : :not_found)
+  end
+
+  def audit_success(privilege, result)
     Audit.logger.log(
       Audit::Event::Check.new(
         user: current_user,
         client_ip: request.ip,
-        resource: resource,
+        resource_id: resource.id,
         privilege: privilege,
-        role: assumed_role,
+        role_id: assumed_role.id,
```

Comment:
`assumed role` can be nil but the in that case an exception will be raised even before and be rescued and audit a failure

Reply:
did you write a test for this scenario? to verify that an audit message is writtent

Reply:
I didnt because even before my changes it was assumed that `assumed_role` is not nil and no audit was written even if it somehow was nil. but anyway, if you think that I need to write that test I will, I just dont really think its necessary

Reply:
@sigalsax once told me that a good boy-scout always leaves the field in a cleaner state than the way he found it :)

will such a test fail if you write it now? do we have an existing bug? if it's simple to fix it, let's do it. if not, please create an issue for this and we can take it in the future.